server.js

```diff
@@ -316,10 +309,12 @@ module.exports.initExpress = function () {
   app.use('/pl/workspace/:workspace_id/container', [
     cookieParser(),
     (req, res, next) => {
+      // Needed for workspaceAuthRouter and `selectAndValidateWorkspace` middleware.
       res.locals.workspace_id = req.params.workspace_id;
       next();
-    }, // needed for workspaceAuthRouter
+    },
     workspaceAuthRouter,
+    require('./middlewares/selectAndValidateWorkspace'),
```

Comment:
Instead of checking the workspace host/state inside the proxy, I instead moved it to separate middleware that runs before the proxy. This gives us greater control over the error code in case we encounter a request bound for a workspace that isn't running.

Reply:
what’s the `originalUrl` key used for here?

Reply:
ah, gotcha! makes sense that `req.url` might get mangled